### PR TITLE
When create a new cr,get the terminating job status as case create error

### PR DESCRIPTION
**Pull request by Mrart**
Add delete false for ownerreference.

### CODE REVIEW

File: controllers/flinkcluster_converter.go [hunk 1]
```diff
@@ -562,7 +562,7 @@ func getDesiredConfigMap(
 			Namespace: clusterNamespace,
 			Name:      configMapName,
 			OwnerReferences: []metav1.OwnerReference{
-				ToOwnerReference(flinkCluster)},
+				ToJobOwnerReference(flinkCluster)},
```
Comment (functicons) [inline controllers/flinkcluster_converter.go:565]: I don't quite understand why we call it `ToJobOwnerReference` in a `ConfigMap` resource? What is the consequence of setting `BlockOwnerDeletion=true` for it? Will I still be able to delete the FlinkCluster CR?

Comment (Mrart) [reply controllers/flinkcluster_converter.go]: commit error

**Comment by functicons** [comment]
Could you explain why we want to set `BlockOwnerDeletion=true` for the job resource? Does that mean the user won't be able to delete a FlinkCluster CR? Then what is the expected way of deleting a job cluster?

**Comment by Mrart** [comment]
@functicons 
   if we set BlockOwnerDeletion=true,  After the job pod delete finished, the flinkcluster cr will deleted. this will fixed the job commit with  the seam name in CR , the new CR will get the old pod status, and the job status turning to failed

**Comment by functicons** [comment]
Did you test what you said in the previous comment? Because I think setting BlockOwnerDeletion=true won't help, because we are setting it for the Job resource (which won't be deleted after the job finishes), not the Pods of the Job resource. In this case, the user has to manually delete the Job resource first, then delete the FlinkCluster. Is my understanding correct?

**Comment by Mrart** [comment]
> Did you test what you said in the previous comment? Because I think setting BlockOwnerDeletion=true won't help, because we are setting it for the Job resource (which won't be deleted after the job finishes), not the Pods of the Job resource. In this case, the user has to manually delete the Job resource first, then delete the FlinkCluster. Is my understanding correct?

I have use this method in our product env, It worked. use BlockOwnerDeletion=true , the cr will deleted after the job resource clean.

**Comment by functicons** [comment]
> the cr will deleted after the job resource clean.

Do you mean you need to manually delete the job resource first?

**Pull request closed without merging**